### PR TITLE
JITX-636: Refactored landpatterns

### PR DESCRIPTION
**Pull request by bpellenard**

### CODE REVIEW

File: utils/land-patterns.stanza [hunk 1]
```diff
@@ -277,19 +285,22 @@ public defn dshape-pth-pad (drill-w:Double, drill-h:Double, pad-w:Double, pad-h:
...
-  layer(SolderMask(Top)) = Circle(C, mask-r)
-  layer(SolderMask(Bottom)) = Circle(C, mask-r)
-  layer(Courtyard(Top)) = Rectangle((drill-r * 2.0), (drill-r * 2.0))
-  layer(Courtyard(Top)) = Rectangle((drill-r * 2.0), (drill-r * 2.0))
+  val circle = Circle(C, mask-r)
+  layer(SolderMask(Top)) = circle
+  layer(SolderMask(Bottom)) = circle
+  val rectangle = Rectangle((drill-r * 2.0), (drill-r * 2.0))
+  layer(Courtyard(Top)) = rectangle
+  layer(Courtyard(Top)) = rectangle
```
Comment (bpellenard) [inline utils/land-patterns.stanza:303]: should not it be `layer(Courtyard(Bottom)) = rectangle`

Comment (PhilippeFerreiraDeSousa) [reply utils/land-patterns.stanza:303]: Oh, we should probably throw an esir error when somebody try to define the same layer twice

Comment (bpellenard) [reply utils/land-patterns.stanza:303]: Is it easy to detect?

**Comment by m-hilgendorf** [comment]
This PR duplicates a lot of the effort in #52 which changes how additional layers are generated.

File: utils/land-patterns.stanza (hunk 1 above)
Comment (m-hilgendorf) [reply utils/land-patterns.stanza:303]: This is not an error in general, you may place different shapes on the same layer.

**Comment by bpellenard** [comment]
Ok, so I am going to close it

**Comment by bpellenard** [comment]
Already done in https://github.com/JITx-Inc/open-components-database/pull/52

**Pull request closed without merging**